pkg/reconciler/autoscaling/hpa/controller.go

```diff
@@ -0,0 +1,117 @@
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	autoscalingv1alpha1 "knative.dev/serving/pkg/apis/autoscaling/v1alpha1"
+	"knative.dev/serving/pkg/autoscaler/config/autoscalerconfig"
+	servingclient "knative.dev/serving/pkg/client/injection/client"
+	metricinformer "knative.dev/serving/pkg/client/injection/informers/autoscaling/v1alpha1/metric"
+	painformer "knative.dev/serving/pkg/client/injection/informers/autoscaling/v1alpha1/podautoscaler"
+	pareconciler "knative.dev/serving/pkg/client/injection/reconciler/autoscaling/v1alpha1/podautoscaler"
+	"knative.dev/serving/pkg/deployment"
+	areconciler "knative.dev/serving/pkg/reconciler/autoscaling"
+	"knative.dev/serving/pkg/reconciler/autoscaling/config"
+
+	kedav1alpha1 "knative.dev/autoscaler-keda/pkg/client/injection/client"
+	keda "knative.dev/autoscaler-keda/pkg/client/injection/informers/keda/v1alpha1/scaledobject"
+)
+
+// NewController returns a new HPA reconcile controller.
+func NewController(
+	ctx context.Context,
+	cmw configmap.Watcher,
+) *controller.Impl {
+	logger := logging.FromContext(ctx)
+	paInformer := painformer.Get(ctx)
+	sksInformer := sksinformer.Get(ctx)
+	hpaInformer := hpainformer.Get(ctx)
+	metricInformer := metricinformer.Get(ctx)
+	kedaInformer := keda.Get(ctx)
+
+	onlyHPAClass := pkgreconciler.AnnotationFilterFunc(autoscaling.ClassAnnotationKey, autoscaling.HPA, false)
+
+	c := &Reconciler{
+		Base: &areconciler.Base{
+			Client:           servingclient.Get(ctx),
+			NetworkingClient: networkingclient.Get(ctx),
+			SKSLister:        sksInformer.Lister(),
+			MetricLister:     metricInformer.Lister(),
+		},
+
+		kubeClient: kubeclient.Get(ctx),
+		kedaLister: kedaInformer.Lister(),
+		kedaClient: kedav1alpha1.Get(ctx),
+		hpaLister:  hpaInformer.Lister(),
+	}
+	impl := pareconciler.NewImpl(ctx, c, autoscaling.HPA, func(impl *controller.Impl) controller.Options {
+		logger.Info("Setting up ConfigMap receivers")
+		configsToResync := []interface{}{
+			&autoscalerconfig.Config{},
+			&deployment.Config{},
+		}
+		resync := configmap.TypeFilter(configsToResync...)(func(string, interface{}) {
+			impl.FilteredGlobalResync(onlyHPAClass, paInformer.Informer())
+		})
+		configStore := config.NewStore(logger.Named("config-store"), resync)
+		configStore.WatchConfigs(cmw)
+		return controller.Options{ConfigStore: configStore}
+	})
+
+	logger.Info("Setting up hpa-class event handlers")
+
+	paInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: onlyHPAClass,
+		Handler:    controller.HandleAll(impl.Enqueue),
+	})
+
+	onlyPAControlled := controller.FilterController(&autoscalingv1alpha1.PodAutoscaler{})
+	handleMatchingControllers := cache.FilteringResourceEventHandler{
+		FilterFunc: pkgreconciler.ChainFilterFuncs(onlyHPAClass, onlyPAControlled),
+		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
+	}
+
+	gk := schema.GroupKind{
+		Group: "keda.sh",
```

Comment:
Are there constants for this in the keda dep?

Reply:
I will check.